src/main/java/org/opensearch/plugin/insights/rules/model/Measurement.java

```diff
@@ -0,0 +1,160 @@
+/*
+ * SPDX-License-Identifier: Apache-2.0
+ *
+ * The OpenSearch Contributors require contributions made to
+ * this file be licensed under the Apache-2.0 license or a
+ * compatible open source license.
+ */
+
+package org.opensearch.plugin.insights.rules.model;
+
+import java.io.IOException;
+import java.util.Locale;
+import java.util.Objects;
+import org.opensearch.core.common.io.stream.StreamInput;
+import org.opensearch.core.common.io.stream.StreamOutput;
+import org.opensearch.core.common.io.stream.Writeable;
+import org.opensearch.core.xcontent.ToXContent;
+import org.opensearch.core.xcontent.ToXContentObject;
+import org.opensearch.core.xcontent.XContentBuilder;
+
+/**
+ * Measurement that is stored in the SearchQueryRecord. Measurement can be of a specific DimensionType and MetricType
+ */
+public class Measurement implements ToXContentObject, Writeable {
+    private static int DEFAULT_COUNT = 1;
+    private DimensionType dimensionType;
+    private MetricType metricType;
+    private Number number;
+    private int count;
+
+    /**
+     * Constructor
+     * @param metricType metricType
+     * @param number number
+     * @param count count
+     * @param dimensionType dimensionType
+     */
+    public Measurement(MetricType metricType, Number number, int count, DimensionType dimensionType) {
+        this.metricType = metricType;
+        this.number = number;
+        this.count = count;
+        this.dimensionType = dimensionType;
+    }
+
+    /**
+     * Constructor
+     * @param metricType metricType
+     * @param number number
+     * @param dimensionType dimensionType
+     */
+    public Measurement(MetricType metricType, Number number, DimensionType dimensionType) {
+        this(metricType, number, DEFAULT_COUNT, dimensionType);
+    }
+
+    /**
+     * Constructor
+     * @param metricType metricType
+     * @param number number
+     */
+    public Measurement(MetricType metricType, Number number) {
+        this(metricType, number, DEFAULT_COUNT, DimensionType.DEFUALT_DIMENSION_TYPE);
+    }
+
+    /**
+     * Add measurement number to the current number based on the dimension type
+     * @param toAdd number to add
+     */
+    public void addMeasurement(Number toAdd) {
+        switch (dimensionType) {
+            case NONE:
+                dimensionType = DimensionType.SUM;
```

Comment:
Makes sense, trying to think of any scenario where we might use the AggregationType = None.
If we cannot think of any will remove NONE and default to SUM.

Reply:
Displaying dimension type = SUM on the API response might be misleading if group_by = NONE since its a singular value and not a sum.
"dimensionType": "NONE" might be more suitable?